### PR TITLE
ZMQ let the testing begin

### PR DESCRIPTION
**Pull request by muscleman**

### CODE REVIEW

**Comment by ArqTras** [comment]
License and other stuff to add :D

**Review by ArqTras — DISMISSED** [review]

File: src/arqma_mq/INotifier.h [hunk 1]
```diff
@@ -26,52 +25,26 @@
 // INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
 // STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
 // THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Parts of this file are copyright (c) 2014-2020 The Arqma Network
```
Comment (malbit) [inline src/arqma_mq/INotifier.h]: Should be “2014-2020, The Monero Project“ instead

File: src/arqma_mq/INotifier.h [hunk 2]
```diff
@@ -26,52 +25,26 @@
...
+// Parts of this file are copyright (c) 2020-2021 Gary Rusher
```
Comment (malbit) [inline src/arqma_mq/INotifier.h]: are we in the future already??
That line is not needed cus Copyright are already stated on top of the file.

File: src/arqma_mq/arqmaMQ.cpp [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+// MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+// THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+// STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+// THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Parts of this file are copyright (c) 2014-2020 The Arqma Network
```
Comment (malbit) [inline src/arqma_mq/arqmaMQ.cpp]: Same as above file :)

File: src/arqma_mq/arqmaMQ.cpp [hunk 4]
```diff
@@ -0,0 +1,210 @@
...
+// Parts of this file are copyright (c) 2020-2021 Gary Rusher
```
Comment (malbit) [inline src/arqma_mq/arqmaMQ.cpp]: same as above file :)

File: src/arqma_mq/arqmaMQ.h [hunk 5]
```diff
@@ -0,0 +1,138 @@
...
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+// MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+// THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+// STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+// THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Parts of this file are copyright (c) 2014-2020 The Arqma Network
```
Comment (malbit) [inline src/arqma_mq/arqmaMQ.h]: Same as above.

File: src/arqma_mq/arqmaMQ.h [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+// Parts of this file are copyright (c) 2020-2021 Gary Rusher
```
Comment (malbit) [inline src/arqma_mq/arqmaMQ.h]: Same as above.

File: src/arqma_mq/utils.hpp [hunk 7]
```diff
@@ -0,0 +1,77 @@
...
+// EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF
+// MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL
+// THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+// SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO,
+// PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+// INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT,
+// STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF
+// THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+//
+// Parts of this file are copyright (c) 2014-2020 The Arqma Network
```
Comment (malbit) [inline src/arqma_mq/utils.hpp]: As above.

File: src/arqma_mq/utils.hpp [hunk 8]
```diff
@@ -0,0 +1,77 @@
...
+// Parts of this file are copyright (c) 2020-2021 Gary Rusher
```
Comment (malbit) [inline src/arqma_mq/utils.hpp]: As above

**Review by malbit — CHANGES_REQUESTED** [review]
I will do tests later on.
License infos need to be fixed.

**Review by ArqTras — APPROVED** [review]
Looks correct and functional.
Additional improvements can be added in future using this as basic.

**Review by ArqTras — APPROVED** [review]

**Review by ArqTras — APPROVED** [review]

**Review by malbit — APPROVED** [review]

**Pull request merged**